src/Keboola/DbExtractor/Extractor/Extractor.php

```diff
@@ -139,20 +139,40 @@ public function export(array $table)
             $query = $table['query'];
         }
 
-        try {
-            /** @var \PDOStatement $stmt */
-            $stmt = $this->executeQuery(
-                $query,
-                isset($table['retries']) ? (int) $table['retries'] : self::DEFAULT_MAX_TRIES
-            );
-        } catch (\Exception $e) {
-            throw $this->handleDbError($e, $table);
-        }
+        $maxTries = isset($table['retries']) ? (int) $table['retries'] : self::DEFAULT_MAX_TRIES;
+        $retryPolicy = new SimpleRetryPolicy($maxTries, ['ErrorException', 'CsvException']);
+        $backOffPolicy = new ExponentialBackOffPolicy(1000);
+        $proxy = new RetryProxy($retryPolicy, $backOffPolicy);
+        $counter = 0;
+        /** @var \Exception $lastException */
+        $lastException = null;
 
-        try {
-            $result = $this->writeToCsv($stmt, $csv, $isAdvancedQuery);
-        } catch (CsvException $e) {
-             throw new ApplicationException("Write to CSV failed: " . $e->getMessage(), 0, $e);
+        $result = $proxy->call(function () use (
+            $maxTries, $query, $table, $outputTable, $isAdvancedQuery, &$counter, &$lastException
+        ) {
+            if ($counter > 0) {
```

Comment:
hmm, would it be simpler if we moved this condition inside the } catch (\Exception $e) { statement?